gettext build fails on macOS due to invoking autoconf

Similar to #488, https://build.macports.org/builders/ports-10.12_x86_64-builder/builds/64404.

```
config.status: creating Makefile
config.status: creating installpaths
config.status: creating po/Makefile
config.status: executing po-directories commands
CDPATH="${ZSH_VERSION+.}:" && cd . && /bin/sh /opt/local/var/macports/build/_opt_bblocal_var_buildworker_ports_build_ports_editors_poedit/poedit/work/poedit-2.1.1/build/Deps.build/Release/_intermediate/gettext/build-aux/missing autoconf
/opt/local/var/macports/build/_opt_bblocal_var_buildworker_ports_build_ports_editors_poedit/poedit/work/poedit-2.1.1/build/Deps.build/Release/_intermediate/gettext/build-aux/missing: line 81: autoconf: command not found
WARNING: 'autoconf' is missing on your system.
         You should only need it if you modified 'configure.ac',
         or m4 files included by it.
         The 'autoconf' program is part of the GNU Autoconf package:
         <http://www.gnu.org/software/autoconf/>
         It also requires GNU m4 and Perl in order to run:
         <http://www.gnu.org/software/m4/>
         <http://www.perl.org/>
make: *** [configure] Error 127
ninja: build stopped: subcommand failed.
Command ./build_deps.sh failed with exit code 1
```

--- a/deps/build.ninja
+++ b/deps/build.ninja
@@ -123,13 +123,13 @@ rule gettext_build
       'note: ⋯  $
       touch `find $
       . -name $
-      configure`' $
+      aclocal.m4`' $
       && $
       tmp=`mktemp` $
       && touch $
       `find . $
       -name $
-      configure` $
+      aclocal.m4` $
       >$$tmp 2>&1 $
       || (cat $
       $$tmp ; $
@@ -139,13 +139,13 @@ rule gettext_build
       'note: ⋯  $
       touch `find $
       . -name $
-      aclocal.m4`' $
+      configure`' $
       && $
       tmp=`mktemp` $
       && touch $
       `find . $
       -name $
-      aclocal.m4` $
+      configure` $
       >$$tmp 2>&1 $
       || (cat $
       $$tmp ; $

--- a/deps/generate_build_ninja.py
+++ b/deps/generate_build_ninja.py
@@ -145,8 +145,8 @@ with open('build.ninja', 'w') as buildfile:
                                  ],
                                  build_commands=[
                                      # Prevent automake regeneration:
-                                     'touch `find . -name configure`',
                                      'touch `find . -name aclocal.m4`',
+                                     'touch `find . -name configure`',
                                      'touch `find . -name config.h.in`',
                                      'touch `find . -name Makefile.in`',
                                      'touch `find . -name *.1`',

